drivers/console/uart_mux.c

```diff
@@ -512,28 +512,11 @@ static int uart_mux_fifo_fill(const struct device *dev,
 	}
 
 	dev_data = DEV_DATA(dev);
-	if (dev_data->dev == NULL) {
+	if (!dev_data->dlci || !dev_data->dev) {
 		return -ENOENT;
 	}
 
-	LOG_DBG("dev_data %p len %d tx_ringbuf space %u",
```

Comment:
Now this `uart_mux_fifo_fill()` probably cannot be called from ISR any more, which we are not doing so no worries there. The problem is that uart_mux is extending the UART API but now they are incompatible as the other requires a call from ISR and the other denies it. So a question, could we add here a check that if we are in ISR, we do the original code, and otherwise call `gsm_dlci_send()` directly?
Also we need to add some comments here why we are doing it like this.

Reply:
Yeah agree